app/src/main/java/com/bottlerocketstudios/mapsdemo/ui/map/YelpViewModel.kt

```diff
@@ -17,26 +21,70 @@ class YelpViewModel : BaseViewModel() {
 
     // UI
     val yelpBusinessState: MutableStateFlow<List<Business>> = MutableStateFlow(emptyList())
-    val dallasLatLng: LatLng = LatLng(LATITUDE, LONGITUDE)
+    val googleMapsMarkersLatLng = yelpBusinessState.map { businessList ->
+        businessList.map { business ->
+            YelpMarker(
+                latitude = business.coordinates.latitude,
+                longitude = business.coordinates.longitude,
+                businessName = business.businessName
+            )
+        }
+    }
+    val selectedMarker: MutableStateFlow<YelpMarker> = MutableStateFlow(YelpMarker(latitude = 0.0, longitude = 0.0, businessName = ""))
+    val dallasLatLng: LatLong = LatLong(DEFAULT_LATITUDE, DEFAULT_LONGITUDE)
+    private var searchJob: Job? = null
 
     private companion object {
-        const val LATITUDE = 32.7767
-        const val LONGITUDE = -96.7970
+        const val DEFAULT_LATITUDE = 32.7767
+        const val DEFAULT_LONGITUDE = -96.7970
+        const val SEARCH_DELAY = 300L
+        const val ZOOM_THRESHOLD = 13
+
+        // Yelp Search Radius are in meters. 40,000 meter is about 25 miles
+        const val MAX_SEARCH_RADIUS = 40000
     }
 
     init {
-        getYelpBusinesses(YelpLatLngSearch(dallasLatLng.latitude, dallasLatLng.longitude))
+        getYelpBusinesses(LatLong(dallasLatLng.latitude, dallasLatLng.longitude), MAX_SEARCH_RADIUS)
     }
 
-    private fun getYelpBusinesses(yelpLatLngSearch: YelpLatLngSearch) {
-        viewModelScope.launch(dispatcherProvider.IO) {
-            yelpRepository.getBusinessesByLatLng(yelpLatLngSearch)
+    private fun getYelpBusinesses(latLong: LatLong, radius: Int?) {
+        launchIO {
+            yelpRepository.getBusinessesByLatLng(latLong, radius)
                 .onSuccess { businessList ->
                     yelpBusinessState.value = businessList
                 }.handleFailure()
         }
     }
+
+    fun getYelpBusinessesOnMapMove(latLong: LatLong, zoomLevel: Float) {
+
+        searchJob?.cancel()
+
+        searchJob = viewModelScope.launch(dispatcherProvider.IO) {
+            // Delay is added to wait for the camera to stop moving before performing
+            // the search.
+            delay(SEARCH_DELAY)
+
+            getYelpBusinesses(
+                latLong = latLong,
+                radius = if (zoomLevel > ZOOM_THRESHOLD) {
+                    null
+                } else {
+                    MAX_SEARCH_RADIUS
+                }
+            )
+        }
+    }
+
     fun resetError() {
         errorStateFlow.value = UserFacingError.NoError
     }
+    fun retrySearch() {
+        getYelpBusinesses(LatLong(dallasLatLng.latitude, dallasLatLng.longitude), radius = null)
```

Comment:
@Josh-Gebbeken @BottleRocket-Colin Should we reset to dallas lat/lng on error or the last known lat/lng sent from the map? If dallas, it would be nice if the map reset back to the dallas lat/lng coordinates.

Reply:
I can easily set that up to be last known coordinates